src/librustc_mir/build/matches/mod.rs

```diff
@@ -112,6 +115,11 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
         }
         self.visibility_scope = outer_source_info.scope;
 
+        if let Some(imaginary_target_block) = opt_imaginary_target_block.take() {
```

Comment:
I think it would be cleaner to use `mem::replace(opt_imaginary_target_block, Some(imaginary_target_block))` here